src/app/Dashboard/Dashboard.tsx

```diff
@@ -36,20 +36,82 @@
  * SOFTWARE.
  */
 import * as React from 'react';
+import { Stack, StackItem } from '@patternfly/react-core';
 import { TargetView } from '@app/TargetView/TargetView';
+import { getFromLocalStorage, saveToLocalStorage } from '@app/utils/LocalStorage';
+import { AddCard } from './AddCard';
+import { DashboardCardActionMenu } from './DashboardCardActionMenu';
 import { AutomatedAnalysisCard } from './AutomatedAnalysis/AutomatedAnalysisCard';
-import { Grid, GridItem } from '@patternfly/react-core';
+
+export interface CardConfig {
+  component: React.FunctionComponent;
+  props?: React.PropsWithChildren<any>;
+}
 
 export interface DashboardProps {}
 
+export interface DashboardCardProps {
+  actions?: JSX.Element[];
+}
+
+export const DashboardCards: CardConfig[] = [
+  {
+    component: AutomatedAnalysisCard,
+    props: {
+      isLarge: true,
+    },
+  },
+];
+
+function getConfigByName(name: String): CardConfig {
+  for (const choice of DashboardCards) {
+    if (choice.component.name === name) {
+      return choice;
+    }
+  }
+  throw new Error(`Unknown card type selection: ${name}`);
+}
+
 export const Dashboard: React.FunctionComponent<DashboardProps> = (props) => {
+  const [cardNames, setCardNames] = React.useState(getFromLocalStorage('DASHBOARD_CFG', []));
+
+  React.useEffect(() => {
+    saveToLocalStorage('DASHBOARD_CFG', cardNames);
+  }, [cardNames]);
+
+  const handleAdd = React.useCallback(
+    (name: string) => {
+      setCardNames((old) => [name, ...old]);
+    },
+    [setCardNames]
+  );
+
+  const handleRemove = React.useCallback(
+    (idx: number) => {
+      setCardNames((old) => {
+        const copy = [...old];
+        copy.splice(idx, 1);
+        return copy;
+      });
+    },
+    [setCardNames]
+  );
+
   return (
     <TargetView pageTitle="Dashboard" compactSelect={false} hideEmptyState>
-      <Grid hasGutter>
-        <GridItem xl2={8} xl={10} lg={10} md={12} sm={12}>
-          <AutomatedAnalysisCard />
-        </GridItem>
-      </Grid>
+      <Stack hasGutter>
+        {cardNames.map(getConfigByName).map((cfg, idx) => (
+          <StackItem key={idx}>
```

Comment:
Since we are aiming to support re-ordering, I think we should not use the index as key, but instead something that is unique the card. Something like `${cardName}-${duplicate-version}` or its hash code?

Reply:
I'll leave that for later since my current definition of the "grid" is really just a reflowing of a stack, but it could become a real grid with x/y coordinates too.